tests/foreman/performance/test_pulp_sync.py

```diff
@@ -0,0 +1,281 @@
+                'raw-sync-{}-clients'.format(current_num_threads)
+            )
+
+            # get max for each iteration
+            for iteration in range(self.sync_iterations):
+                total_max_timing[current_num_threads].append(max([
+                    subtest_dict.get('thread-{}'.format(thread))[iteration]
+                    for thread in range(current_num_threads)
+                ]))
+
+        self.logger.debug(
+            'Total Results for all tests from 2 threads to 10 threads: {}'
+            .format(total_max_timing)
+        )
+        self._write_stat_pulp_concurrent(total_max_timing)
+
+    def _write_raw_csv_chart_pulp(
+            self,
+            raw_file_name,
+            time_result_dict,
+            current_num_threads,
+            test_case_name):
+        """Write csv and chart for raw data of Pulp Test sync/resync"""
+        self.logger.debug(
+            'Timing result is: {}'.format(time_result_dict)
+        )
+
+        with open(raw_file_name, 'a') as handler:
+            writer = csv.writer(handler)
+            writer.writerow([test_case_name])
+
+            # for each thread, write its head and data
+            for i in range(len(time_result_dict)):
+                writer.writerow(time_result_dict.get('thread-{}'.format(i)))
+            writer.writerow([])
+
+        # generate line chart of raw data
+        test_category = self._get_output_filename(raw_file_name)
+        generate_line_chart_raw_pulp(
+            time_result_dict,
+            'Pulp Synchronization Raw Timings Line Chart - '
+            '({0}-{1}-clients)'
+            .format(test_category, current_num_threads),
+            '{0}-{1}-clients-raw-data-line-chart.svg'
+            .format(test_category, current_num_threads)
+        )
+
+    def _write_stat_pulp_concurrent(self, total_max_timing):
+        """Compute statistics on concurrent sync data across tests"""
+
+        with open(self.stat_file_name, 'a') as handler:
+            writer = csv.writer(handler)
+            writer.writerow(['concurrent-sync-stat'])
+            writer.writerow(['test-case', 'min', 'median', 'max', 'std'])
+
+        stat_dict = {}
+        for test in range(2, len(total_max_timing) + 2):
+            time_list = total_max_timing.get(test)
+            stat_tuple = generate_stat_for_pulp_sync(
+                test,
+                time_list,
+                self.stat_file_name,
+            )
+            # insert each returned tuple of statistics into stat_dict
+            stat_dict.update({test: stat_tuple})
+
+        generate_line_chart_stat_pulp(
+            stat_dict,
+            'Satellite 6 Pulp Concurrent Sync Test',
+            'perf-statistics-concurrent-sync.svg',
+            self.max_num_tests
+        )
+
+    def test_sequential_synchronization(self):
+        """
+        @Test: Synchronize two repos sequentially
+
+        @Steps:
+        1. get list of all enabled repositories (setUpClass)
```

Comment:
Add a blank line between `@Steps:` and the beginning of the list, like so:

    @Steps:

    1. get list of all enabled repositories (setUpClass)